mio/process/frame_helper.py

```diff
@@ -0,0 +1,280 @@
+"""
+This module contains a helper class for frame operations.
+It should be organized in a different way to make it more readable and maintainable.
+"""
+
+from typing import Tuple
+
+import cv2
+import numpy as np
+
+from mio import init_logger
+from mio.models.process import FreqencyMaskingConfig, NoisePatchConfig
+
+logger = init_logger("frame_helper")
+
+
+class NoiseDetectionHelper:
+    """
+    Helper class for noise detection and frame processing.
+    """
+
+    def __init__(self, height: int, width: int):
+        """
+        Initialize the FrameProcessor object.
+        Block size/buffer size will be set by dev config later.
+
+        Parameters:
+        height (int): Height of the video frame.
+        width (int): Width of the video frame.
+
+        Returns:
+        NoiseDetectionHelper: A NoiseDetectionHelper object
+        """
+        self.height = height
+        self.width = width
+
+    def split_by_length(self, array: np.ndarray, segment_length: int) -> list[np.ndarray]:
```

Comment:
something like this seems pretty general purpose, but we don't have a good place to put 'general array manipulation functions' like a utils junk drawer would be. i like this helper class idea, bundling some operations together with some staticmethods. i get where this is coming from being split off from the previous implementation, so i'm fine with this, but in the future i might want to play with this helper class idea, make it its own module, and make some generic "ArrayHelper"/"BufferHelper" kind of classes

Reply:
Yeah, that does make sense. After all, this whole package is about tinkering with numpy arrays not just this denoising module.